### PR TITLE
Added "Meetup.com" provider

### PR DESCRIPTION
**Pull request by dance2die**
Hi, I have added a provider for Meetup.com.

Would you review the content and see if it's mergeable?
Please let me know should the merge does not conform to a guideline or a standard.

## Test Steps
1. Add  `.AddMeetup` in `Startup.cs`

```csharp
.AddMeetup(options =>
{
    options.ClientId = "CLIENT ID";
    options.ClientSecret = "CLIENT SECRET";
    options.Scope.Add("basic");
});
```

> The list of Scope values can be found [here](https://www.meetup.com/meetup_api/auth/#oauth2-scopes).

2. Create a Meetup OAuth consumer from [here](https://secure.meetup.com/meetup_api/oauth_consumers/create) to get a `ClientId` (shown as `Key`) and `ClientSecret` (shown as `Secret`) 
2.1 and set the Redirect URL to "https://localhost:44318/"
firefox_2018-02-23_20-42-11

> Once you've added the Meetup provider, the test site will display "Connect using Meetup" as shown below
chrome_2018-02-23_20-35-51

3. Login to Meetup.com

4. Redirected page shows claims processed
chrome_2018-02-23_20-38-51

### CODE REVIEW

**Comment by dance2die** [comment]
@PinpointTownes Hi Kevin,

Would you be able to guide me on what I am doing wrong and how I can fix the issue?
(I've left a comment on the build failure on Gitter, just in case)

Thank you.

**Comment by kinosang** [comment]
CI failed because it can't download `KoreBuild` from GitHub. No idea why it happened.

File: src/AspNet.Security.OAuth.Meetup/MeetupAuthenticationOptions.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            ClaimActions.MapJsonKey("urn:meetup:status", "status");
+            ClaimActions.MapJsonKey("urn:meetup:lat", "lat");
+            ClaimActions.MapJsonKey("urn:meetup:lon", "lon");
+            ClaimActions.MapJsonKey("urn:meetup:joined", "joined");
+            ClaimActions.MapJsonKey("urn:meetup:city", "city");
+            ClaimActions.MapJsonKey("urn:meetup:country", "country");
+            ClaimActions.MapJsonKey("urn:meetup:localized_country_name", "localized_country_name");
+            ClaimActions.MapJsonKey("urn:meetup:state", "state");
+
+			// Map user's photo information returned as a nested object
```
Comment (kinosang) [inline src/AspNet.Security.OAuth.Meetup/MeetupAuthenticationOptions.cs]: keep the indent of 12 spaces.

File: src/AspNet.Security.OAuth.Meetup/MeetupAuthenticationOptions.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            AuthorizationEndpoint = MeetupAuthenticationDefaults.AuthorizationEndpoint;
+            TokenEndpoint = MeetupAuthenticationDefaults.TokenEndpoint;
+            UserInformationEndpoint = MeetupAuthenticationDefaults.UserInformationEndpoint;
+
+            ClaimActions.MapJsonKey(ClaimTypes.NameIdentifier, "id");
+            ClaimActions.MapJsonKey(ClaimTypes.Name, "name");
+            ClaimActions.MapJsonKey(ClaimTypes.Email, "email");
+
+			// Map custom claims
+            ClaimActions.MapJsonKey("urn:meetup:name", "name");
```
Comment (kinosang) [inline src/AspNet.Security.OAuth.Meetup/MeetupAuthenticationOptions.cs]: See #216, it introduces `ClaimTypes` constants for the claims.

Comment (dance2die) [reply src/AspNet.Security.OAuth.Meetup/MeetupAuthenticationOptions.cs]: OK, let me refer to the #216

File: src/AspNet.Security.OAuth.Meetup/MeetupAuthenticationOptions.cs (hunk 1 above)
Comment (dance2die) [reply src/AspNet.Security.OAuth.Meetup/MeetupAuthenticationOptions.cs]: I am sorry about that.
My Visual Studio has tabs as a space by default. I will update the indentations.

**Comment by dance2die** [comment]
@kinosang 
AppVeyor is still failing while Travis is passing.

According to GitHub [issue](https://github.com/aspnet/KoreBuild/issues/223#issuecomment-318732440) on AspNet KoreBuild, zip version is not supported any more

**Comment by kevinchalet** [comment]
> According to GitHub issue on AspNet KoreBuild, zip version is not supported any more

We'll probably move to the new version at some point, but the error you're seeing on AppVeyor doesn't have anything to do with that: it's caused by the fact the GitHub.com folks removed support for TLS 1.0 and 1.1 2 days ago. The AppVeyor team is working on a global fix to force .NET apps to use TLS 1.2 in all their CI images.

In the meantime, let's just ignore the AppVeyor errors.

**Comment by dance2die** [comment]
Thank you for looking into the AppVeyor issue @PinpointTownes.

**Comment by dance2die** [comment]
I've rebased according to an advice in #224.
Would you review to see if the rebase is done properly? (My first time doing a rebase).
Thank you.

File: src/AspNet.Security.OAuth.Meetup/MeetupClaimTypes.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        public const string PhotoHighResolutionLink = "urn:meetup:photo.highres_link";
+
+        public const string PhotoLink = "urn:meetup:photo.photo_link";
+
+        public const string PhotoThumbnailLink = "urn:meetup:photo.thumb_link";
+
+        public const string PhotoBaseUrl = "urn:meetup:photo.base_url";
+
+        public const string PhotoType = "urn:meetup:photo.type";
+
```
Comment (martincostello) [inline src/AspNet.Security.OAuth.Meetup/MeetupClaimTypes.cs:43]: Nit: remove blank line here.

**Review by martincostello — COMMENTED** [review]
This needs re-syncing with the dev branch so as to not include the changes to the build scripts.

**Comment by martincostello** [comment]
Thanks for the PR @dance2die - as part of the preparation for ASP.NET Core 3.0 support, tests are being added to help make things easier to maintain and validate going forwards (see #292).

Once tests are merged into the dev branch, could you copy the approach to add tests for the new provider into this PR please?

**Comment by martincostello** [comment]
#280 has been merged to `dev` now. If you rebase and update any tests as appropriate, we can look at merging this PR soon.

**Comment by martincostello** [comment]
Closing due to inactivity and merge conflicts - please re-open and rebase if you would like to pursue adding a provider for Meetup.com in a future release.

**Comment by dance2die** [comment]
@martincostello 
Sorry about the late follow up.

I haven't been using .NET core for awhile and I will leave it to others to do a PR should this is need Meetup provider.

**Pull request closed without merging**